master/internal/task/allocation.go

```diff
@@ -406,6 +425,31 @@ func (a *Allocation) Cleanup(ctx *actor.Context) {
 	}
 }
 
+func (a *Allocation) getPorts(ctx *actor.Context) (int, int, int, int, error) {
+	dTrainport, err := portregistry.GetPort(dtrainSSHPortBase)
```

Comment:
knowing which ports to we need should happen at a higher level, probably where we build the trial/command specs. GetPort should happen here, but it should probably look like
```
for _, portName := range taskspec.ListOfPortNames struct {

}
```